sdk/include/stdlib.h

```diff
@@ -85,6 +85,15 @@ DEFINE_ALLOCATOR_CAPABILITY(__default_malloc_capability, MALLOC_QUOTA)
  */
 #define MALLOC_CAPABILITY STATIC_SEALED_VALUE(__default_malloc_capability)
 
+/**
+ * Define how long a call to `malloc` and `calloc` can block to fulfil an
+ * allocation. Regardless of this value, `malloc` and `calloc` will only ever
+ * block to wait for the quarantine to be processed. This means that, even with
+ * a non-zero value of `MALLOC_WAIT_TICKS`, `malloc` would immediately return
+ * if the heap or the quota is exhausted.
+ */
+#define MALLOC_WAIT_TICKS 30
```

Comment:
Maybe wrap this in an `#ifndef` so that users can override it?  Since our malloc is always-inline, it's safe to make this a per-compilation-unit thing.